### ISSUE TITLE
Cannot force i965 driver on X11

### ISSUE THREAD

**Issue by jbeich** [intel/libva#355]
Regressed by 6650b390d1c21c. Works fine on KMS console and Wayland but fails on X11. Forcing `iHD` (via `LIBVA_DRIVER_NAME`) and `i965` auto-detection (via `iHD` absence) works fine.
```
$ LIBVA_DRIVER_NAME=i965 vainfo
libva info: VA-API version 1.6.0
libva info: User environment variable requested driver 'i965'
libva info: Trying to open /usr/local/lib/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_1_6
libva error: /usr/local/lib/dri/i965_drv_video.so init failed
libva info: va_openDriver() returns -1
vaInitialize failed with error code -1 (unknown libva error),exit
```
According to `VA_INTEL_DEBUG=1` init function fails due to [DRI2 check](https://github.com/intel/intel-vaapi-driver/blob/329975c63123610fc750241654a3bd18add75beb/src/intel_driver.c#L110-L113).

**Comment by jbeich** [comment]
`__vaDriverInit_1_6` fails due to [DRI2 check](https://github.com/intel/intel-vaapi-driver/blob/329975c63123610fc750241654a3bd18add75beb/src/intel_driver.c#L110-L113)

**Comment by XinfengZhang** [comment]
see, va_DRI2GetDriverName also call authentication , previous logic , it will call va_getDriverName firstly

**Comment by XinfengZhang** [comment]
@jbeich I filed #356 , could you help to try it?

**Comment by jbeich** [comment]
I confirm, #356 fixes the issue.

**Comment by uartie** [comment]
#356 also fixes a segfault regression introduced by #351 when using gstreamer-vaapi+iHD
